### PR TITLE
Update sync to 2.0.0-rc25

### PR DESCRIPTION
**Pull request by beeender**

### CODE REVIEW

**Comment by cmelchior** [comment]
```
io.realm.ManagedOrderedRealmCollectionTests > deleteLastFromRealm[MANAGED_REALMLIST][SHIELD Tablet K1 - 5.1.1] FAILED 
Test failed to run to completion. Reason: 'Instrumentation run failed due to 'Native crash''. Check device logcat for details
Tests on SHIELD Tablet K1 - 5.1.1 failed: Instrumentation run failed due to 'Native crash'
:realm-library:connectedObjectServerDebugAndroidTest FAILED
```

Seems to be crashing consistently. I'm investigating now.

**Comment by cmelchior** [comment]
Minimal unittest https://github.com/realm/realm-java/pull/5346/files#diff-42daba2ae36de5b442c65c55b9c8877eR2126

Seems like something is going wrong when looking up non-ascii field names.

**Comment by Zhuinden** [comment]
You can try adding some case for `árvíztűrőtükörfúrógép` it checks every accent available in hungarian 😄

**Comment by cmelchior** [comment]
Some more hunting

In https://github.com/realm/realm-java/blob/master/realm/realm-library/src/main/cpp/io_realm_internal_SharedRealm.cpp#L65

The returned schema does not seem to contain the correct information. It already reports the wrong Index here, which points to a problem in Object Store.

But this unit test in Object Store checks out fine :/

```

    SECTION("Create Non-latin names on when creating Realm") {
        TestFile config;
        config.schema_version = 1;
        config.schema = Schema{
                {"RødGrødMedFløde", {
                    {"델타", PropertyType::Int},
                    {"Δέλτα", PropertyType::Int},
                    {"베타", PropertyType::Int},
                    {"βήτα", PropertyType::Int},
                    {"æøå", PropertyType::Int},
                }}
        };

        auto realm = Realm::get_shared_realm(config);
        REQUIRE(realm->schema().size() == 1);
        auto obj_schema = realm->schema().find("RødGrødMedFløde");
        REQUIRE(obj_schema->property_for_name("델타")->table_column == 0);
        REQUIRE(obj_schema->property_for_name("Δέλτα")->table_column == 1);
    }
```

**Comment by Zhuinden** [comment]
Hmm. Does it only happen with sync realm, or with regular mobile database too?

**Comment by cmelchior** [comment]
```
void ObjectStore::set_schema_columns(Group const& group, Schema& schema)
{
    for (auto& object_schema : schema) {
        auto table = table_for_object_schema(group, object_schema);
        if (!table) {
            continue;
        }
        for (auto& property : object_schema.persisted_properties) {
            property.table_column = table->get_column_index(property.name);
        }
    }
}
```
https://github.com/realm/realm-object-store/blob/master/src/object_store.cpp#L777

Tracked it down to this method in Object Store. On `armeabi-v7a` it does the wrong thing, on `x86` it works 😢

**Comment by cmelchior** [comment]
Updated the unit test so crash is now:
```
09-29 21:47:20.644 27990-27990/? A/DEBUG: ABI: 'arm'
09-29 21:47:20.644 27990-27990/? A/DEBUG: pid: 26636, tid: 27241, name: roidJUnitRunner  >>> io.realm.test <<<
09-29 21:47:20.644 27990-27990/? A/DEBUG: signal 5 (SIGTRAP), code 1 (TRAP_BRKPT), fault addr 0xd40eb028
09-29 21:47:20.644 27990-27990/? A/DEBUG:     r0 00000000  r1 00000001  r2 00000002  r3 00000000
09-29 21:47:20.644 27990-27990/? A/DEBUG:     r4 e6c4e008  r5 00000005  r6 ffffffff  r7 ffffffff
09-29 21:47:20.644 27990-27990/? A/DEBUG:     r8 d4c790e4  r9 d4c7927c  sl d4c7928c  fp d4c79004
09-29 21:47:20.644 27990-27990/? A/DEBUG:     ip 00000000  sp d4c78fa0  lr d4127070  pc d40eb028  cpsr 200f0010
09-29 21:47:20.744 27990-27990/? A/DEBUG: backtrace:
09-29 21:47:20.745 27990-27990/? A/DEBUG:     #00 pc 002d2028  /data/app/io.realm.test-rAMuPwPUcEnc2etXDH1jeQ==/lib/arm/librealm-jni.so (_ZN5realm11ObjectStore20apply_schema_changesERNS_5GroupEyRNS_6SchemaEyNS_10SchemaModeERKSt6vectorINS_12SchemaChangeESaIS7_EESt8functionIFvvEE+180)
09-29 21:47:20.745 27990-27990/? A/DEBUG:     #01 pc 002e4f0c  /data/app/io.realm.test-rAMuPwPUcEnc2etXDH1jeQ==/lib/arm/librealm-jni.so (_ZN5realm5Realm13update_schemaENS_6SchemaEySt8functionIFvSt10shared_ptrIS0_ES4_RS1_EES2_IFvS4_EEb+956)
09-29 21:47:20.745 27990-27990/? A/DEBUG:     #02 pc 002ff430  /data/app/io.realm.test-rAMuPwPUcEnc2etXDH1jeQ==/lib/arm/librealm-jni.so (_ZN5realm5_impl16RealmCoordinator9get_realmENS_5Realm6ConfigE+2216)
09-29 21:47:20.745 27990-27990/? A/DEBUG:     #03 pc 002e0e94  /data/app/io.realm.test-rAMuPwPUcEnc2etXDH1jeQ==/lib/arm/librealm-jni.so (_ZN5realm5Realm16get_shared_realmENS0_6ConfigE+80)
09-29 21:47:20.745 27990-27990/? A/DEBUG:     #04 pc 0025093c  /data/app/io.realm.test-rAMuPwPUcEnc2etXDH1jeQ==/lib/arm/librealm-jni.so (Java_io_realm_internal_SharedRealm_nativeGetSharedRealm+100)
09-29 21:47:20.745 27990-27990/? A/DEBUG:     #05 pc 00026b73  /data/app/io.realm.test-rAMuPwPUcEnc2etXDH1jeQ==/oat/arm/base.odex (offset 0x21000)
```

Not entirely sure why my Nexus 6P reports `arm` as ABI 🤔

File: realm/realm-library/src/androidTest/java/io/realm/RealmObjectTests.java [hunk 1]
```diff
@@ -2125,9 +2125,17 @@ public void setter_string_long_values() {
...
-        realm.beginTransaction();
-        NonLatinFieldNames obj = realm.createObject(NonLatinFieldNames.class);
-        obj.setΔέλτα(0);
-        realm.commitTransaction();
+        RealmConfiguration config = configFactory.createConfigurationBuilder().name("greekWTF").schema(NonLatinFieldNames.class).build();
+        Realm realm = Realm.getInstance(config);
+        try {
+            realm.beginTransaction();
+            RealmObjectSchema schema = realm.getSchema().get(NonLatinFieldNames.class.getSimpleName());
+            schema.getFieldType("델타");
```
Comment (Zhuinden) [inline realm/realm-library/src/androidTest/java/io/realm/RealmObjectTests.java]: If you call `schema.getFieldType()` on a different field, then the setter will crash?? O_O

Comment (cmelchior) [reply realm/realm-library/src/androidTest/java/io/realm/RealmObjectTests.java]: No, it crashes there. I just kept the rest for further test.

**Comment by cmelchior** [comment]
Adding the same code to one of our example projects makes it run fine :( The example runs with a release build).

In debug mode changing `-0g` to `-O0/O1/O2` still crashes it.

**Comment by cmelchior** [comment]
Seems to be parsing now. Fast merging to unblock other testing, but keep track of https://github.com/realm/realm-java/issues/5354 for investigating this. 

We still don't have any reason to believe this is affecting release builds, only debug builds.

**Pull request merged**